docs/cicd_insider/README.md

```diff
@@ -207,9 +199,43 @@ The scripts are based on a cross-platform TeamsFx command line tool [TeamsFx-CLI
 
 > Note: To enable `@microsoft/teamsfx-cli` running in CI mode, turn on `CI_ENABLED` by `export CI_ENABLED=true`. In CI mode, `@microsoft/teamsfx-cli` is friendly for CI/CD.
 
-Please keep in mind that you need to set Azure and M365 credentials in your environment variables safely. For example if you are using Github as your source code repository, you can use the [Github Secrets](https://docs.github.com/en/actions/reference/encrypted-secrets) to securely store your environment variables.
+Please keep in mind that you must store Azure service principal credentials in your environment variables securely. For example if you are using Github as your source code repository, you can use the [Github Secrets](https://docs.github.com/en/actions/reference/encrypted-secrets) to securely store your credentials.
+
+The following table gives details about the required environment variables used in the script.
+
+|Name|Description|
+|---|---|
+|AZURE_SERVICE_PRINCIPAL_NAME|The service principal name of Azure used to provision resources.|
+|AZURE_SERVICE_PRINCIPAL_PASSWORD|The password of Azure service principal.|
+|AZURE_SUBSCRIPTION_ID|To identify the subscription in which the resources will be provisioned.|
+|AZURE_TENANT_ID|To identify the tenant in which the subscription resides.|
+
+To create Azure service principals for use, refer to [here](#how-to-create-azure-service-principals-for-use).
+
+# How to create Azure service principals for use?
+To provision and deploy resources targeting Azure inside CI/CD, you must create an Azure service principal for use.
+
+Briefly, the steps include:
+1. Register an Azure AD application in single tenant, and it requires sufficient permissions in your Azure AD tenant.
+2. Assign a role to your Azure AD application to access your Azure subscription, and `Contributor` role is recommended. 
+3. Create a new Azure AD application secret.
+4. Grab your tenant id, application id(AZURE_SERVICE_PRINCIPAL_NAME), and the secret(AZURE_SERVICE_PRINCIPAL_PASSWORD) for use.
+
+For detailed guidelines, refer to [the official document](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal). There're three ways to create service principal, [Azure portal](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal), [PowerShell](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-authenticate-service-principal-powershell), [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli), and you can choose the way you like.
+
+# How to publish Teams app using Teams dev portal? 
+If there's any changes related to Teams app's manifest file, you may want to publish the Teams app again to update the manifest.
+
+To publish Teams app manually, you may leverage [Teams' dev portal](https://dev.teams.microsoft.com/home).
```

Comment:
It would be better to use `Developer Portal for Teams`

Reply:
agree, update it soon.